### PR TITLE
change save posts flow

### PR DESCRIPTION
**Pull request by yuda1124**
This work is fixing bug that there are no change in favorite button when you change category in listView. I fount the reason of that problem is saving favorite information as state. So, I linked favorite information directly. card component is used very many part of our application. plz review carefully.

### CODE REVIEW

File: shared/reducers/list.js [hunk 1]
```diff
@@ -36,6 +36,13 @@ const list = (state = initialState, action) => {
     return update(state, {
       currentPostedUri: { $set: action.uri }
     });
+  case types.toggleSaved:
+    let eventIndex = state.dataSource.findIndex((post) => {
```
Comment (ovekyc) [inline shared/reducers/list.js]: const

Comment (ovekyc) [inline shared/reducers/list.js]: You can just write it like
const eventIndex = state.dataSource.dinfIndex(post => post.key === action.eventKey);

Comment (ochanje210) [inline shared/reducers/list.js]: find*

File: shared/actions/listActions.js [hunk 2]
```diff
@@ -99,17 +99,21 @@ export const getDetailImage = (key) => {
...
       return fetch(address, {
         method: 'GET',
         headers
       })
       .then(response => response.json())
-      .then(json =>
-        dispatch(receiveImages(json))
-      );
+      .then(json => {
+        return new Promise((resolve) => {
```
Comment (ochanje210) [inline shared/actions/listActions.js]: Is there a point of returning new promise here when you can simply return and make it thenable?

Comment (yuda1124) [inline shared/actions/listActions.js]: Can you explain me more?

Comment (yuda1124) [inline shared/actions/listActions.js]: oh I got it. look at the 121 line in activityCard component. it is used to find image index equal to imageKey.

Comment (yuda1124) [inline shared/actions/listActions.js]: I changed this another pr.

Comment (ochanje210) [inline shared/actions/listActions.js]: ok! thanks

Comment (yuda1124) [inline shared/actions/listActions.js]: so give me approved!!!!!!!!!!!!!

Comment (ochanje210) [inline shared/actions/listActions.js]: lol

**Pull request merged**